hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/JetPlan.java

```diff
@@ -478,23 +480,87 @@ public SqlResult execute(QueryId queryId, List<Object> arguments) {
         }
     }
 
-    static class SelectOrSinkPlan extends JetPlan {
+    abstract static class DmlPlan extends JetPlan {
+        DmlPlan(PlanKey planKey) {
+            super(planKey);
+        }
+    }
+
+    static class SinkPlan extends DmlPlan {
+        private final Set<PlanObjectKey> objectKeys;
+        private final QueryParameterMetadata parameterMetadata;
+        private final DAG dag;
+        private final JetPlanExecutor planExecutor;
+        private final List<Permission> permissions;
+
+        SinkPlan(
+                PlanKey planKey,
+                QueryParameterMetadata parameterMetadata,
+                Set<PlanObjectKey> objectKeys,
+                DAG dag,
+                JetPlanExecutor planExecutor,
+                List<Permission> permissions
+        ) {
+            super(planKey);
+
+            this.objectKeys = objectKeys;
+            this.parameterMetadata = parameterMetadata;
+            this.dag = dag;
+            this.planExecutor = planExecutor;
+            this.permissions = permissions;
+        }
+
+        QueryParameterMetadata getParameterMetadata() {
+            return parameterMetadata;
+        }
+
+        DAG getDag() {
+            return dag;
+        }
+
+        @Override
+        public boolean isCacheable() {
+            return !objectKeys.contains(PlanObjectKey.NON_CACHEABLE_OBJECT_KEY);
+        }
+
+        @Override
+        public boolean isPlanValid(PlanCheckContext context) {
+            return context.isValid(objectKeys);
+        }
+
+        @Override
+        public void checkPermissions(SqlSecurityContext context) {
+            for (Permission permission : permissions) {
+                context.checkPermission(permission);
+            }
+        }
+
+        @Override
+        public boolean producesRows() {
+            return false;
+        }
+
+        @Override
+        public SqlResult execute(QueryId queryId, List<Object> arguments) {
+            return planExecutor.execute(this, queryId, arguments);
+        }
+    }
+
+    static class SelectPlan extends DmlPlan {
```

Comment:
DML = `INSERT/UPDATE/DELETE`, not `SELECT`.
The `DmlPlan` doesn't contain anything, maybe we can delete it?